src/dodal/beamlines/i22.py

```diff
@@ -41,338 +39,267 @@
-        wait_for_connection,
-        fake_with_ophyd_sim,
+@instantiation_behaviour(
+    default_use_mock_at_connection=True, default_timeout_for_connect=10
+)
+def panda1():
+    """Create HDFPanda instance for panda1."""
+    return HDFPanda(
+        name="panda1",
+        prefix="-EA-PANDA-01:",
         directory_provider=get_directory_provider(),
     )
 
 
 @skip_device()
-def panda2(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> HDFPanda:
-    return device_instantiation(
-        HDFPanda,
-        "panda2",
-        "-EA-PANDA-02:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
+def panda2():
+    """Create HDFPanda instance for panda2."""
+    return HDFPanda(
+        name="panda2",
+        prefix="-EA-PANDA-02:",
         directory_provider=get_directory_provider(),
     )
 
 
 @skip_device()
-def panda3(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> HDFPanda:
-    return device_instantiation(
-        HDFPanda,
-        "panda3",
-        "-EA-PANDA-03:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
+def panda3():
+    """Create HDFPanda instance for panda3."""
+    return HDFPanda(
+        name="panda3",
+        prefix="-EA-PANDA-03:",
         directory_provider=get_directory_provider(),
     )
 
 
 @skip_device()
-def panda4(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> HDFPanda:
-    return device_instantiation(
-        HDFPanda,
-        "panda4",
-        "-EA-PANDA-04:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
+def panda4():
+    """Create HDFPanda instance for panda4."""
+    return HDFPanda(
+        name="panda4",
+        prefix="-EA-PANDA-04:",
         directory_provider=get_directory_provider(),
     )
 
 
-def oav(
-    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
-) -> AravisDetector:
-    return device_instantiation(
-        NXSasOAV,
-        "oav",
-        "-DI-OAV-01:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
+@instantiation_behaviour(
+    default_use_mock_at_connection=True, default_timeout_for_connect=10
+)
+def oav():
+    """Create OAV instance with specific settings."""
+    metadata_holder = NXSasMetadataHolder(
+        x_pixel_size=(3.45e-3, "mm"),  # Double check this figure
+        y_pixel_size=(3.45e-3, "mm"),
+        description="AVT Mako G-507B",
+        distance=(-1.0, "m"),
+    )
+
+    return NXSasOAV(
+        name="oav",
+        prefix="-DI-OAV-01:",
         drv_suffix="DET:",
         hdf_suffix="HDF5:",
-        metadata_holder=NXSasMetadataHolder(
-            x_pixel_size=(3.45e-3, "mm"),  # Double check this figure
-            y_pixel_size=(3.45e-3, "mm"),
-            description="AVT Mako G-507B",
-            distance=(-1.0, "m"),
-        ),
+        metadata_holder=metadata_holder,
         directory_provider=get_directory_provider(),
     )
 
 
 @skip_device()
```

Comment:
Could: Is `@instantiation_behaviour` now rich enough that we can replace calls to `@skip_device` with it?

Reply:
yes, that is the premise of the function signature, not 100% sure about the implementation details still

Reply:
ok it is not a supported feature at the moment.

working on it now